blocks/pipeline.py

```diff
@@ -106,10 +121,10 @@ def save_remote(self) -> None:
                 model.save_remote()
 
     def children(self) -> List[Element]:
-        return self.datasource.children() + [self] + [self.models]
+        return self.datasource_fit.children() + [self] + [self.models]
 
     def get_hierarchy(self) -> Hierarchy:
-        source_hierarchy = self.datasource.get_hierarchy()
+        source_hierarchy = self.datasource_fit.get_hierarchy()
```

Comment:
hmm, why is it datasource_fit here, and not datasource_predict?
an arbitrary choice? what if these two are out of sync?

Reply:
It's an arbitrary choice right now, will fix this later!

Reply:
I have pushed a commit now resolving this!

Reply:
It's not good nevertheless, trying to fix it.